src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
First off: I don't have strong feelings about this, I'm just giving my experience with deprecation messages. 

So: the deprecation message should of course mention the alternative - that's the most important bit! And for callers, that (plus the deprecation version) is the only information that is relevant.

The justification is relevant for people who work on improving the library and addressing tech debt. So it should be in a @todo. Why do you think it is critical to give the justification in the deprecation message?

(Again, for the record: I think this is a minor point. An interesting detail discussion about how to write perfect documentation. Shouldn't block a patch.)